enterprise/server/remote_execution/runner/runner.go

```diff
@@ -540,22 +526,23 @@ func (r *commandRunner) cleanupCIRunner(ctx context.Context) error {
 	return res.Error
 }
 
-type ContainerProvider func(ctx context.Context, props *platform.Properties, st *repb.ScheduledTask, state *rnpb.RunnerState, workDir string) (*container.TracedCommandContainer, error)
+type ContainerProvider interface {
+	NewContainer(context.Context, *platform.Properties, *repb.ScheduledTask, *rnpb.RunnerState, string) (container.CommandContainer, error)
```

Comment:
nit: there are quite a few args to this func, maybe pack into a struct? (if it's too much though we can always do this later)

Reply:
I will defer it to later, mostly because I am fairly bad at naming things and we already have a bunch of `*Options` structs.